app/models/service_template.rb

```diff
@@ -45,6 +45,19 @@ class ServiceTemplate < ApplicationRecord
   virtual_has_one :custom_actions, :class_name => "Hash"
   virtual_has_one :custom_action_buttons, :class_name => "Array"
 
+  # This method creates
+  def self.create_catalog_item(options, auth_user)
+    create(options.except(:config_info)).tap do |service_template|
```

Comment:
I would include the whole create process in a `transaction` block because multiple db records are created in this method and they should be rollback back if any error occurs.